src/main/scala/firrtl/passes/ExpandWhens.scala

```diff
@@ -200,11 +220,13 @@ object ExpandWhens extends Pass {
     * @todo Preserve Info
     * @note Remove IsInvalids on attached Analog-typed components
     */
-  private def expandNetlist(netlist: Netlist, attached: Set[WrappedExpression]) =
+  private def expandNetlist(netlist: Netlist, attached: Set[WrappedExpression], sourceInfoMap: InfoMap) =
     netlist map {
       case (k, WInvalid) => // Remove IsInvalids on attached Analog types
         if (attached.contains(k)) EmptyStmt else IsInvalid(NoInfo, k.e1)
-      case (k, v) => Connect(NoInfo, k.e1, v)
+      case (k, v) =>
+        val info = sourceInfoMap.getOrElse(k.e1.serialize, NoInfo)
```

Comment:
Since you're giving it a default value, does it need `getOrElse(..., NoInfo)` here?

Reply:
Good catch